src/init.cpp

```diff
@@ -1134,7 +1126,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     // see Step 2: parameter interactions for more information about these
     fListen = GetBoolArg("-listen", DEFAULT_LISTEN);
     fDiscover = GetBoolArg("-discover", true);
-    fNameLookup = GetBoolArg("-dns", true);
+    fNameLookup = GetBoolArg("-dns", fNameLookup);
```

Comment:
Please use a constant for this, instead of passing through the old value of the global.